### PR TITLE
BosonVoucher - add transferPremintedFrom

### PR DESCRIPTION
**Pull request by zajck**
TransferPremintedFrom is non-standart method to transfer preminted tokens. It is more efficient than generic transferFrom methods, since it does not have to perform binary search.
To make that possible, user must also supply `offerId` as input argument. To get `offerId` from `tokenId`, user can call `getPreMintStatus`, which was made public in this PR.

### CODE REVIEW

File: test/protocol/clients/BosonVoucherTest.js [hunk 1]
```diff
@@ -1136,7 +1136,7 @@ describe("IBosonVoucher", function () {
     });
   });
 
-  context("Token transfers", function () {
+  context.only("Token transfers", function () {
```
[the remark this answers is not in the document]
Comment (zajck) [reply test/protocol/clients/BosonVoucherTest.js]: Fixed.

File: test/protocol/clients/BosonVoucherTest.js [hunk 2]
```diff
@@ -1414,6 +1414,13 @@ describe("IBosonVoucher", function () {
               await expect(
                 bosonVoucher.connect(operator)[selector](operator.address, rando.address, tokenId, ...additionalArgs)
               ).to.be.revertedWith(RevertReasons.ERC721_CALLER_NOT_OWNER_OR_APPROVED);
+
+              // It should also fail if trasnfer done with transferPremintedFrom
```
[the remark this answers is not in the document]
Comment (zajck) [reply test/protocol/clients/BosonVoucherTest.js]: Fixed.

File: test/protocol/clients/BosonVoucherTest.js [hunk 3]
```diff
@@ -1467,6 +1474,195 @@ describe("IBosonVoucher", function () {
...
+        });
+
+        it("Transfer preminted voucher, which was not committed but burned already", async function () {
+          // Void offer
+          await offerHandler.connect(operator).voidOffer(offerId);
+
+          // Burn preminted vouchers
+          await bosonVoucher.connect(operator).burnPremintedVouchers(offerId);
+
+          // None of reserverd but not preminted tokens should have an owner
```
Comment (anajuliabit) [inline test/protocol/clients/BosonVoucherTest.js:1633]:
```suggestion
          // None of reserved but not preminted tokens should have an owner
```

**Review by anajuliabit — CHANGES_REQUESTED** [review]
just a small typo

File: test/protocol/clients/BosonVoucherTest.js (hunk 3 above)
Comment (zajck) [reply test/protocol/clients/BosonVoucherTest.js]: Fixed here and in other places with same mistake

**Review by anajuliabit — APPROVED** [review]

**Review by mischat — APPROVED** [review]
LGTM

**Pull request merged**